### DISCUSSION TITLE
[일지] 6월 10일 (목) 오후 회의록

### DISCUSSION THREAD

**Discussion by olenmg** [bcaitech1/p4-opt-5-vibrhanium-#73] [일지]
### 코드
- 베이스라인 작성 예정 (종헌)
- Knowledge Distillation 예정 (아름)

```python
from src.utils.common import calc_macs 

before = calc_macs(model, (3,224,224))
model = pruning(model)
after = calc_macs(model, (3,224,224))
```

### 멘토님께 질문 (오늘 8시)
- nlp 논문 추천 ~~ㅋㅋ~~
- NLP 모델에서 경량화? 
    - 특히 transformer encoder의 self-attention module의 경량화를 어떻게 하는지
- nlp에서 gnn 활용 사례
- NLP에서 적절한 input 을 찾는 방법
- 서빙 관점에서 ML/DL 관련 외 필요한 지식이나 참고할만한 자료
- 인공지능 개발자 면접에서 지원자가 갖춰야할 역량

### 기준 모델 선정
VGG-16, ResNet-34 둘다 50 epoch
이따가 ipynb 코드 전달

### 해볼 것 (대회 관련)
- tensor decomposition 의 심화
- automl로 찾은 모델(student)에 지식증류 하기 (knowledge distilation 적용할 teacher모델 찾기)
- efficientNetb0 pytorch pruning 적용 + 10강 내용 적용
- 이미지 방향 정렬
- shufflenet을 더 경량화할 수 있는가? => 이미지 사이즈는 늘리고.
- distilliation
    1. pretrained model(fine-tuning X)으로 student B에 distillation한 이후 B를 fine-tuning
    2. pretrained model + fine-tuning으로 student B에 distilation (O) => student B를 미리 fine-tuning해놓고 distillation
- augmentation 기법, 새로운 criterion 등 image classification 자체에 적용할 기법들

### 계획
- 6/11(금) 오전까지: ResNet/VGG 돌려서 Baseline 맞춰서 기능구현
    - 기능 구현을 했을 때 shufflenet.yaml, mobilenetv3.yaml를 불러와서 적용했을 때 문제없이 돌아가야함
- 6/11(금) 자정까지: 실험 다 해두고 기록 (F1/ACC, parameter 수 변화, MACs 변화, 속도)
- 주말에는 구현된 메소드들로 모델 실제로 경량화
    - optuna로 categorical 만들어서 경량화 기법 적용
    - quantization에서 global pruning 혹은 부분부분 hyper parameter 활용하여 pruning
- 월요일에 augmentation, loss 같은거 찾기, 그나마 최적화된 모델 찾기
- 화요일은 못해본거, 남은거

**Comment by areum514** [comment]
기초논문 추천
        - [Convolutional Neural Networks for Sentence Classification](https://arxiv.org/abs/1408.5882) 구현 추천
        - [cnn-lstm]( https://arxiv.org/abs/1508.06615) 구현 추천
        - [seq2seq with attention](https://arxiv.org/abs/1409.0473)
        - [Attention Is All You Need](https://papers.nips.cc/paper/2017/file/3f5ee243547dee91fbd053c1c4a845aa-Paper.pdf)(면접에서 자주 출제, 주석 내용 물어보는 경우 多)
        - [BERT](https://arxiv.org/abs/1810.04805)
